ios/FluentUI/Command Bar/CommandBar.swift

```diff
@@ -0,0 +1,290 @@
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+/**
+ `CommandBar` is a horizontal scrollable list of icon buttons divided by groups.
+ Provide `itemGroups` in `init` to set the buttons in the scrollable area. Optional `leadingItem` and `trailingItem` add fixed buttons in leading and trailing positions. Each `CommandBarItem` will be represented as a button.
+ Set the `delegate` property to determine whether a button can be selected and deselected, and listen to selection changes.
+ */
+@objc(MSFCommandBar)
+open class CommandBar: UIView {
+    // Hierarchy:
+    //
+    // leadingButton
+    // containerView
+    // |--layer.mask -> containerMaskLayer (fill containerView)
+    // |--subviews
+    // |  |--scrollView (fill containerView)
+    // |  |  |--subviews
+    // |  |  |  |--stackView
+    // |  |  |  |  |--buttons (fill scrollView content)
+    // trailingButton
+
+    // MARK: - Public methods
+
+    public init(itemGroups: [CommandBarItemGroup], leadingItem: CommandBarItem? = nil, trailingItem: CommandBarItem? = nil) {
+        self.itemGroups = itemGroups
+
+        super.init(frame: .zero)
+
+        if let leadingItem = leadingItem {
+            self.leadingButton = button(forItem: leadingItem, isFixed: true)
+        }
+        if let trailingItem = trailingItem {
+            self.trailingButton = button(forItem: trailingItem, isFixed: true)
+        }
+
+        translatesAutoresizingMaskIntoConstraints = false
+
+        configureHierarchy()
+        updateButtonsState()
+    }
+
+    @available(*, unavailable)
+    public required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    /// Apply `isEnabled` and `isSelected` state from `CommandBarItem` to the buttons
+    @objc public func updateButtonsState() {
+        for button in itemsToButtonsMap.values {
+            button.updateState()
+        }
+    }
+
+    // MARK: Public properties
+
+    @objc public weak var delegate: CommandBarDelegate?
+
+    // MARK: Overrides
+
+    public override var intrinsicContentSize: CGSize {
+        .zero
+    }
+
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+
+        containerMaskLayer.frame = containerView.bounds
+        updateShadow()
+    }
+
+    // MARK: - Private properties
+
+    private let itemGroups: [CommandBarItemGroup]
+
+    private lazy var itemsToButtonsMap: [CommandBarItem: CommandBarButton] = {
+        let allButtons = itemGroups.flatMap({ $0 }).map({ button(forItem: $0) }) +
+            [leadingButton, trailingButton].compactMap({ $0 })
+
+        return Dictionary(uniqueKeysWithValues: allButtons.map { ($0.item, $0) })
+    }()
+
+    // MARK: Views and Layers
+
+    private lazy var containerView: UIView = {
+        let containerView = UIView()
+        containerView.translatesAutoresizingMaskIntoConstraints = false
+        containerView.layer.mask = containerMaskLayer
```

Comment:
why is there a need for container view? why can't this be applies directly to the view?

Reply:
The mask layer is to apply the gradient effect on the sides of the scroll view. Applying directly to `CommandBar` would corp the sides of the view. And also it can't be applied to the scroll view since the coordinate is different.